fitbenchmarking/jacobian/analytic_jacobian.py

```diff
@@ -39,9 +38,11 @@ def eval(self, params, **kwargs):
             jac = jac / e[:, None]
         elif self.problem.options.cost_func_type == "hellinger_nlls":
             # calculates the Jacobian of the hellinger(root) NLLS cost function
-            jac = jac/(2*sqrt(self.problem.eval_model(params, x=x)[:, None]))
+            jac = jac/(2*np.sqrt(self.problem.eval_model(params, x=x)
+                       [:, None]))
         elif self.problem.options.cost_func_type == "poisson":
             jac = jac * (1 - y / self.problem.eval_model(params, x=x))[:, None]
```

Comment:
This could be having issues as it's only accurate when `y[i] != 0`...

For the `y[i] == 0` case, it will be:

```
cf=self.problem.eval_model(x=x, params=params)
jac=jac
```

So we could try using a mask like in the `_safe_a_log_b` function in `poisson_cost_func.py`.


I also see what you were saying with the penalise negative values discontinuity... I guess we could try setting the jacobian to 0 if something does take a step that ends up with a negative value (although as far as I know, none of the methods take steps towards a worse result!)